Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1624,7 +1624,7 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Event>> Document::create_event(StringView i
     } else if (Infra::is_ascii_case_insensitive_match(interface, "focusevent"sv)) {
         event = UIEvents::FocusEvent::create(realm, FlyString {});
     } else if (Infra::is_ascii_case_insensitive_match(interface, "hashchangeevent"sv)) {
-        event = Event::create(realm, FlyString {}); // FIXME: Create HashChangeEvent
+        event = HTML::HashChangeEvent::create(realm, FlyString {}, {});
```

Comment:
thanks for catching this! it seems there must be the same problem for at least `popstate` event.

Reply:
`createEvent()` is an old legacy API, and it doesn't even know about `popstate` according to the spec :)